File: test/benchmark/beng/python/mandelbrot.py
#!/usr/bin/env python3
# BENG Benchmark: mandelbrot (Python)
# XOR checksum approach
import sys
import time

N = int(sys.argv[1]) if len(sys.argv) > 1 else 200

t0 = time.perf_counter_ns()
checksum = 0
for y in range(N):
    bits = 0
    bit_num = 0
    for x in range(N):
        cr = 2.0 * x / N - 1.5
        ci = 2.0 * y / N - 1.0
        zr = 0.0
        zi = 0.0
        inside = 1
        for _ in range(50):
            tr = zr * zr - zi * zi + cr
            ti = 2.0 * zr * zi + ci
            zr, zi = tr, ti
            if zr * zr + zi * zi > 4.0:
                inside = 0
                break
        bits = (bits << 1) | inside
        bit_num += 1
        if bit_num == 8:
            checksum ^= bits
            bits = 0
            bit_num = 0
    if bit_num > 0:
        bits <<= (8 - bit_num)
        checksum ^= bits
t1 = time.perf_counter_ns()
print(checksum)
print(f"__TIMING__:{(t1 - t0) / 1e6:.3f}")

File: test/benchmark/beng/js/mandelbrot.js
// BENG Benchmark: mandelbrot (Node.js reference)
// XOR checksum approach (matches Lambda adaptation)

const N = parseInt(process.argv[2] || "500");

const __t0 = process.hrtime.bigint();
let checksum = 0;
for (let y = 0; y < N; y++) {
    let bits = 0, bitNum = 0;
    for (let x = 0; x < N; x++) {
        let cr = 2.0 * x / N - 1.5;
        let ci = 2.0 * y / N - 1.0;
        let zr = 0.0, zi = 0.0;
        let inside = 1;
        for (let i = 0; i < 50; i++) {
            const tr = zr * zr - zi * zi + cr;
            const ti = 2.0 * zr * zi + ci;
            zr = tr; zi = ti;
            if (zr * zr + zi * zi > 4.0) { inside = 0; break; }
        }
        bits = (bits << 1) | inside;
        bitNum++;
        if (bitNum === 8) {
            checksum ^= bits;
            bits = 0; bitNum = 0;
        }
    }
    if (bitNum > 0) {
        bits <<= (8 - bitNum);
        checksum ^= bits;
    }
}
const __t1 = process.hrtime.bigint();
console.log(checksum);
process.stdout.write("__TIMING__:" + Number(__t1 - __t0) / 1e6 + "\n");
